src/impl.cpp

```diff
@@ -484,8 +484,9 @@ void Manifold::Impl::CreateHalfedges(const Vec<ivec3>& triProp,
       const int pair1 = ids[k];
       Halfedge& h1 = halfedge_[pair1];
       if (h0.startVert != h1.endVert || h0.endVert != h1.startVert) break;
-      if (halfedge_[NextHalfedge(pair0)].endVert ==
-          halfedge_[NextHalfedge(pair1)].endVert) {
+      if (h1.pairedHalfedge != kRemovedHalfedge &&
```

Comment:
The bug is that we still match removed halfedges when checking for opposed triangles. This now fixes the issue.

Reply:
Nice, I thought I tried this, but I must have gotten it wrong.